icu4j/main/tests/core/src/com/ibm/icu/dev/test/text/SpoofCheckerTest.java

```diff
@@ -470,6 +481,21 @@ void checkSkeleton(SpoofChecker sc, int type, String input, String expected, Str
         assertEquals(testName + " test at line " + lineNumberOfTest + " :  Expected (escaped): " + expected, uExpected, actual);
     }
 
+    // Internal function to run a single skeleton test case.
+    //
+    // Run a single confusable skeleton transformation test case.
+    //
+    void checkBidiSkeleton(SpoofChecker sc, int direction, String input, String expected, String testName) {
+        String uInput = Utility.unescape(input);
```

Comment:
Could we fold some of these lines together. Most of them do not need the on stack String. Also, why you need to create t here? junit print these out for you automatically if the assertEquals failed